src/FsCodec.SystemTextJson/UnionConverter.fs

```diff
@@ -25,3 +41,120 @@ module private Union =
                 caseConstructor = cases |> Array.map (fun c -> FSharpValue.PreComputeUnionConstructor(c, true))
             } |> Some
     let tryGetUnion : Type -> Union option = memoize _tryGetUnion
+
+
+    /// Paralells F# behavior wrt how it generates a DU's underlying .NET Type
+    let inline isInlinedIntoUnionItem (t : Type) =
+        t = typeof<string>
+        //|| t.IsValueType
```

Comment:
🤔 if you have a union of int * int, is this not needed? (no need to respond/explain, have not looked at full code)

Reply:
Yeah I think you're right. I think what this actually needs is a check for JsonElement for when we're keeping a JsonElement field, see https://github.com/jet/FsCodec/pull/43#discussion_r408609791